### PR TITLE
Remove libdrm_amdgpu and other unnecessary code 

### PR DESCRIPTION
**Pull request by evelikov**
This series removes the a bunch of dead or no longer needed code. In particular:

 - libdrm_amdgpu has two better alternatives in-tree - individual and the gpu_metrics sysfs nodes
   - bonus now we properly track/open only the required ones
- ifdef spaghetty in keybinds code
- the find_folder() API - we can use ls() instead
- excessive pipelines when fetching basic data

The vendor/device handling is still a bit iffy - I might be sorting that out next.

Note: this has been only lightly tested, so please give it a spin on your end and report issues.

### CODE REVIEW

**Comment by evelikov** [comment]
Been using this for a few days without issues. Just some 600+ loc removed

@flightlessmango @jackun any takers?

**Comment by Faalagorn** [comment]
I gave it a quick try on my RX 580, so far no issues at the first glance, I'll keep on running with that to see if I encounter any issues down the road.

**Comment by flightlessmango** [comment]
It looks good to me, I would just like @jackun to sign off as well

File: src/overlay.cpp [hunk 1]
```diff
@@ -639,57 +638,35 @@ void init_gpu_stats(uint32_t& vendorID, uint32_t reported_deviceID, overlay_para
...
+               amdgpu.memory_clock = fopen((path + dir + "/freq2_input").c_str(), "r");
             if (!amdgpu.temp)
-               amdgpu.temp = fopen((path + tempFolder + "/temp1_input").c_str(), "r");
+               amdgpu.temp = fopen((path + dir + "/temp1_input").c_str(), "r");
             if (!amdgpu.power_usage)
-               amdgpu.power_usage = fopen((path + tempFolder + "/power1_average").c_str(), "r");
+               amdgpu.power_usage = fopen((path + dir + "/power1_average").c_str(), "r");
 
-            vendorID = 0x1002;
             break;
```
Comment (jackun) [inline src/overlay.cpp:664]: Tangling `break`?

**Comment by jackun** [comment]
Do Vega APUs support `gpu_metrics`, and why libdrm etc? At least Vega64 dgpu doesn't seem to.

File: src/overlay.cpp [hunk 2]
```diff
@@ -639,57 +638,35 @@ void init_gpu_stats(uint32_t& vendorID, uint32_t reported_deviceID, overlay_para
...
             SPDLOG_DEBUG("Using gpu_metrics");
          }
-#ifdef HAVE_LIBDRM_AMDGPU
-         else {
-            int idx = -1;
-            //TODO make neater
-            int res = sscanf(path.c_str(), "/sys/class/drm/card%d", &idx);
-            std::string dri_path = "/dev/dri/card" + std::to_string(idx);
-
-            if (!params.enabled[OVERLAY_PARAM_ENABLED_force_amdgpu_hwmon] && res == 1 && amdgpu_open(dri_path.c_str())) {
```
Comment (jackun) [inline src/overlay.cpp:649]: Then no need for `force_amdgpu_hwmon` param

File: src/overlay.cpp (hunk 1 above)
Comment (evelikov) [reply src/overlay.cpp:664]: Yup should be after the `}` - fix coming in a moment.

**Comment by evelikov** [comment]
> Do Vega APUs support `gpu_metrics` (a.k.a why libdrm etc used in first place)? At least Vega64 dgpu doesn't seem to. e: vega12, vega20, renoir, vangogh, navi10, navi21(?), MI cards, maybe more... so only APUs since 2020?

AFAICT the standalone nodes - gpu_busy_percent, temp1_input and others, are present on all GPUs. So even if the composite `gpu_metrics` node is missing, all GPU will still have basic data.

That said, looks like I've forgot a check - currently the `gpu_metrics` node will be omitted if `gpu_busy_percent` is missing. It seems very unlikely to happen, but still not the right thing to do. Will add extra patch on top.

File: src/overlay.cpp (hunk 2 above)
Comment (evelikov) [reply src/overlay.cpp:649]: Indeed - will drop it.

**Comment by jackun** [comment]
IIRC `gpu_busy_percent` gave "Invalid Argument" on Vega 3 etc. unless patched since https://github.com/flightlessmango/MangoHud/issues/53

**Comment by evelikov** [comment]
Sounds like something which should be reported and fixed in the amdgpu driver. Adding/maintaining a complete code-path for such a bug that does not sound palatable IMHO.

**Review by jackun — APPROVED** [review]

**Comment by evelikov** [comment]
Doing a quick test shows that something has gone off.
Quickly flipping to draft - so this doesn't go in and break the world

**Comment by evelikov** [comment]
The `gpu_busy_percent check` that the last commit removed had a _very_ misleading meaning - it was used to distinguish between `card0` and `card0-DP-1` nodes. See the commit description (and inline comment) for details.

**Comment by jackun** [comment]
> to distinguish between the card node (cardX) and the
card output node (cardX-output-foo).

More of a is using `amdgpu` check but filters those (cardX-output-foo) out too :P

**Comment by evelikov** [comment]
Fwiw there's another dozen or so patches series sitting on top of this work - https://github.com/flightlessmango/MangoHud/pull/707

**Comment by jackun** [comment]
APUs need newer (v)bioses, I guess: https://github.com/torvalds/linux/blob/d888c83fcec75194a8a48ccd283953bdba7b2550/drivers/gpu/drm/amd/pm/powerplay/hwmgr/smu10_hwmgr.c#L1291

**Pull request merged**